From 947788aec862c3854c0d49da85d7e5ecf108d922 Mon Sep 17 00:00:00 2001
From: Kuba Schenk <abukcz@gmail.com>
Date: Thu, 1 Dec 2016 21:48:26 +0100
Subject: [PATCH 29/49] OMS7-N: ApplicationsState: add filter for Substratum
 icon overlays [1/2]

This commit allows the framework to handle the filtering of the icon overlays found for OMS.

Base this off the work from @KreAch3R

Change-Id: I594c993977733e67f566ac65df50ad2e1bbdbdd3
(cherry picked from commit 4d682464550f71e72e491934c78b8a42fdfc0348)
---
 .../android/settingslib/applications/ApplicationsState.java   | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
index 6fecd4044ca..d3a6e217735 100644
--- a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
+++ b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
@@ -1325,6 +1325,17 @@ public class ApplicationsState {
         }
     };
 
+    public static final AppFilter FILTER_SUBSTRATUM_ICONS = new AppFilter() {
+        public void init() {
+        }
+
+        @Override
+        public boolean filterApp(AppEntry entry) {
+            return !((entry.info.metaData != null) &&
+                    (entry.info.metaData.getString("Substratum_IconPack") != null));
+        }
+    };
+
     public static final AppFilter FILTER_WORK = new AppFilter() {
         private int mCurrentUser;
 
-- 
2.11.1

